torch/distributed/autograd/__init__.py

```diff
@@ -0,0 +1,25 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import torch
+
+if hasattr(torch._C, "_dist_autograd_init") and not torch._C._dist_autograd_init():
+    raise RuntimeError("Failed to initialize PyTorch distributed autograd support")
+
+class context:
+    '''
+        Autograd context object to wrap forward and backward passes when using
+        distributed autograd.
```

Comment:
Since this API isn't needed in normal PyTorch usage, it's good to briefly explain why this is needed (and what the consequences are of getting this wrong). Also, in the docs here, you say "forward and backward passes", but then in the example below you include the optimizer step. Should the optimizer step be included?